src/promptflow-core/promptflow/executor/_service/utils/batch_coordinator.py

```diff
@@ -83,7 +83,7 @@ async def exec_line(self, request: LineExecutionRequest):
     def exec_aggregation(self, request: AggregationRequest):
         """Execute aggregation nodes for the batch run."""
         with self._flow_executor._run_tracker.node_log_manager:
-            aggregation_result = self._flow_executor._exec_aggregation(
+            aggregation_result = self._flow_executor.exec_aggregation(
```

Comment:
What is the difference between `exec_aggregation` and `_exec_aggregation`? Do we need this logic in the server scenario?
![image](https://github.com/microsoft/promptflow/assets/111329184/870ddd78-0099-42f1-b5aa-6882f974a059)

Btw, `PythonExecutorProxy` also has the same codes.
![image](https://github.com/microsoft/promptflow/assets/111329184/bdc9514c-b9b4-4866-988e-415dbd1b39aa)

Reply:
1. Do validation for inputs
2. Apply default value
3. Convert flow input type

So, I believe we should do above logic for each case.